apps/meteor/client/views/root/hooks/useUserSync.ts

```diff
@@ -0,0 +1,72 @@
+import type { UserStatus } from '@rocket.chat/core-typings';
+import { useConnectionStatus, useUserId, useUserPreference, useMethod } from '@rocket.chat/ui-contexts';
+import { useQuery } from '@tanstack/react-query';
+import { UserPresence } from 'meteor/rocketchat:user-presence'; // Still needed for UserPresence API
+import { Session } from 'meteor/session';
```

Comment:
BTW I just remembered we have a hook called `useSessionDispatch` to be used instead of this.